src/Documenter.jl

```diff
@@ -462,6 +465,11 @@ function deploydocs(;
                     # Remove the unencrypted private key.
                     isfile(keyfile) && rm(keyfile)
                 end
+
+                # Undo ssh config changes.
```

Comment:
It turns out that you can customize git's SSH command using the [`GIT_SSH` variable](https://git-scm.com/book/en/v2/Git-Internals-Environment-Variables#Miscellaneous). We could `mktemp` the `ssh_config` and set `GIT_SSH` to `ssh -F tmp_ssh_config` perhaps (I haven't tried it though, so not sure if it works).